### ISSUE TITLE
Fargate/ECS stack for back-end deployment

### ISSUE THREAD

**Issue by chriswilty** [ScottLogic/prompt-injection#385]
Use Typescript-CDK to define a stack for deploying the back-end container to AWS. To begin with, we will deploy the stack manually.

CDK example:
https://docs.aws.amazon.com/cdk/v2/guide/ecs_example.html

Node container and other Fargate/ECS/ELB considerations:
https://medium.com/@arliber/aws-fargate-from-start-to-finish-for-a-nodejs-app-9a0e5fbf6361

**Notes**
- #379 must be completed first!
- What to do about CORS before our UI stack is deployed? Do we allow all, or allow localhost:5173 (vite's devserver default port)?
- Until we assess and tighten the security / roles of the deployed resources, we should remove (destroy) the deployed stage every time after testing. This is trivial with CDK.

**Questions**
- Do we want to start with a shared, hard-coded API key (injected into the environment at stack synthesis) or adopt secrets manager from the start?
- Do we want to set strict rate limits from the start, for safety?
- Do we want at least two instances to begin with, to verify load balancing?

**Out of Scope**
- UI stack will be defined in a separate issue
- Pipeline deployment of both stacks will be covered in a subsequent issue

**Acceptance Criteria**
CDK code for (manually) deploying a VPC / ELB / Fargate / ECS stack.
Deployed, verified then destroyed using CDK CLI.

**Comment by chriswilty** [comment]
**Progress to date**

I have two flavours of API stack fronted by API Gateway: one using just ALB with HTTP API, and one using NLB => ALB with REST API. The former is better as the backend is actually receiving headers to represent the (secure) origin, just that they are the "wrong" headers. With the latter, we don't receive any headers for the origin, so the backend cannot trust the connection and does not send the secure cookie.

Going forwards, I will split this issue into three:
- this one, for the Fargate stack
- Route53 (domain with cert manager) for fronting the secure, Application-Load-Balanced API
- authentication with Cognito (which needs UI authentication first)
